OpenRA.Game/Traits/World/ActorMap.cs

```diff
@@ -534,5 +534,16 @@ public IEnumerable<Actor> ActorsInWorld()
 		{
 			return bins.SelectMany(bin => bin.Actors.Where(actor => actor.IsInWorld));
 		}
+
+		public IEnumerable<Actor> GetActorsOnRay(CPos start, CPos end)
```

Comment:
Remove the `.Where(...)`, do `var actors = GetUnitsAt(pos)`, and then add an `if (actors.Any())` before the second `foreach` and enumerate over `actors`.